psd-web/spec/features/sign_in_spec.rb

```diff
@@ -66,6 +72,35 @@ def expect_incorrect_email_or_password
     expect(page).to have_css("#otp_code-error", text: "Error: Incorrect security code")
   end
 
+  scenario "user signs in with correct secondary authentication code after requesting a second code" do
+    visit "/sign-in"
+    fill_in_credentials
+
+    first_2fa_code = otp_code
+    expect_user_to_have_received_sms_code(first_2fa_code)
+
+    expect(page).to have_css("h1", text: "Check your phone")
+
+    click_link "Not received a text message?"
+
+    expect(page).to have_css("h1", text: "Resend security code")
+    expect(page).to have_current_path("/text-not-received")
+
+    click_on "Resend security code"
+
+    second_2fa_code = otp_code
+    expect(second_2fa_code).not_to eq first_2fa_code
+    expect_user_to_have_received_sms_code(second_2fa_code)
+
+    expect(page).to have_css("h1", text: "Check your phone")
+
+    fill_in "Enter security code", with: otp_code
+    click_on "Continue"
```

Comment:
Might be worth changing this to also assert that it's a button being clicked (`clicked_on` works for both buttons and links):

```suggestion
    click_button "Continue"
```